app/src/main/java/dev/arkbuilders/arkmemo/ui/viewmodels/GraphicNotesViewModel.kt

```diff
@@ -35,6 +33,8 @@ class GraphicNotesViewModel @Inject constructor(): ViewModel() {
     private val editPaths = ArrayDeque<DrawPath>()
 
     private var svg = SVG()
+    private val svgLiveData = MutableLiveData<SVG>()
+    val observableSvgLiveData = svgLiveData as LiveData<SVG>
```

Comment:
@shubertm Kotlin's `StateFlow` API is a good choice here, to reduce the dependency of the app on Android platform API. [1]

However, since this is a simple use-case, if we use `StateFlow` here, we'll have to write more boiler plate code to observe the drawing event, as below:

```
viewLifecycleOwner.lifecycleScope.launch {
            viewLifecycleOwner.lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
                graphicNotesViewModel.drawState.collectLatest {
                    enableSaveText(it?.getPaths()?.isNotEmpty() == true)
                }
            }
        }
```

The original logic which uses `LiveData`:

```
graphicNotesViewModel.observableSvgLiveData.observe(viewLifecycleOwner) {
            enableSaveText(it.getPaths().isNotEmpty())
        }
```

Is there any other reason for which you suggest changing to `StateFlow`, apart from [1]?

Reply:
No, no other good reason. Let's keep it around then